### PR TITLE
test:  Add new test for multinode: two pods on two nods have different IPS

### LINKED ISSUE #9925: add multinode integration test two pods should get two different ips on different nodes
add a subtest

deploy an app with more than 1 replica on multi node
their pods ips should be different

https://github.com/medyagh/minikube/blob/fc10bca4aec2bf7b53357ad93729643db519fd64/test/integration/multinode_test.go#L29

### PR DESCRIPTION
**Pull request by hetong07**
Fix #9925 

Add a subtest for deploying an app on a two node cluster and verifies their Pod IPs are different. Also check both pods could resolve to kubernetes official website per [this request](https://github.com/kubernetes/minikube/issues/9925#issuecomment-743824065).

**Before PR**:
No such test.

**After PR**:
```
=== RUN   TestMultiNodeWithDeployment
=== RUN   TestMultiNodeWithDeployment/serial
=== RUN   TestMultiNodeWithDeployment/serial/FreshStart2Nodes
    multinode_test.go:104: (dbg) Run:  out/minikube start -p minikube --wait=true --memory=2200 --nodes=2 -v=8 --alsologtostderr 
    multinode_test.go:104: (dbg) Done: out/minikube start -p minikube --wait=true --memory=2200 --nodes=2 -v=8 --alsologtostderr : (3m1.806122796s)
    multinode_test.go:110: (dbg) Run:  out/minikube -p minikube status --alsologtostderr
=== RUN   TestMultiNodeWithDeployment/serial/DeployApp2Nodes
    multinode_test.go:423: (dbg) Run:  out/minikube kubectl -p minikube -- apply -f ./testdata/multinodes/hello-deployment.yaml
    multinode_test.go:428: (dbg) Run:  out/minikube kubectl -p minikube -- rollout status deployment/hello
    multinode_test.go:428: (dbg) Done: out/minikube kubectl -p minikube -- rollout status deployment/hello: (7.12904332s)
    multinode_test.go:434: (dbg) Run:  out/minikube kubectl -p minikube -- get pods -o jsonpath='{.items[*].status.podIP}'
    multinode_test.go:447: (dbg) Run:  out/minikube kubectl -p minikube -- get pods -o jsonpath='{.items[*].metadata.name}'
    multinode_test.go:455: (dbg) Run:  out/minikube kubectl -p minikube -- exec hello-f45cbcf6d-tjflv -- nslookup 8.8.8.8
    multinode_test.go:455: (dbg) Run:  out/minikube kubectl -p minikube -- exec hello-f45cbcf6d-zmqp7 -- nslookup 8.8.8.8
=== CONT  TestMultiNodeWithDeployment
    helpers_test.go:171: Cleaning up "minikube" profile ...
    helpers_test.go:174: (dbg) Run:  out/minikube delete -p minikube
    helpers_test.go:174: (dbg) Done: out/minikube delete -p minikube: (5.361721499s)
--- PASS: TestMultiNodeWithDeployment (197.50s)
    --- PASS: TestMultiNodeWithDeployment/serial (192.13s)
        --- PASS: TestMultiNodeWithDeployment/serial/FreshStart2Nodes (182.62s)
        --- PASS: TestMultiNodeWithDeployment/serial/DeployApp2Nodes (9.51s)
PASS

```

### CODE REVIEW

File: test/integration/multinode_test.go [hunk 1]
```diff
@@ -387,3 +408,44 @@ func validatNameConflict(ctx context.Context, t *testing.T, profile string) {
...
+	if err != nil {
+		t.Errorf("failed retrieve Pod IPs")
+	}
+	podIPs := strings.Split(strings.Trim(rr.Stdout.String(), "'"), " ")
+	if len(podIPs) != 2 {
+		t.Errorf("expected 2 Pod IPs but got %d", len(podIPs))
+	}
+	if podIPs[0] == podIPs[1] {
+		t.Errorf("expected 2 different pod IPs but got %s and %s", podIPs[0], podIPs[0])
+	}
```
Comment (sadlil) [inline test/integration/multinode_test.go:435]: If possible verify pods in node are in the the `node.PodCIDR` range.

File: test/integration/multinode_test.go [hunk 2]
```diff
@@ -387,3 +408,44 @@ func validatNameConflict(ctx context.Context, t *testing.T, profile string) {
...
+	// get Pod names
+	rr, err = Run(t, exec.CommandContext(ctx, Target(), "kubectl", "-p", profile, "--", "get", "pods", "-o", "jsonpath='{.items[*].metadata.name}'"))
+	if err != nil {
+		t.Errorf("failed get Pod names")
+	}
+	podNames := strings.Split(strings.Trim(rr.Stdout.String(), "'"), " ")
+
+	// verify both Pods could resolve a public DNS
+	for _, name := range podNames {
+		_, err = Run(t, exec.CommandContext(ctx, Target(), "kubectl", "-p", profile, "--", "exec", name, "--", "nslookup", "8.8.8.8"))
```
Comment (sadlil) [inline test/integration/multinode_test.go:446]: Verify from inside the pod, we are also able to resolve external domain ie `google.com` and any internal service domain `<svc>.<namespace>.svc.cluster.local`.

File: test/integration/testdata/multinodes/hello-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: hello
```
Comment (sadlil) [inline test/integration/testdata/multinodes/hello-deployment.yaml:4]: Should we rename the deployment to something like `multi-node-pod-ip-test`

File: test/integration/testdata/multinodes/hello-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      affinity:
+        # ⬇⬇⬇ This ensures pods will land on separate hosts
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions: [{ key: app, operator: In, values: [hello-from] }]
+            topologyKey: "kubernetes.io/hostname"
+      containers:
+      - name: hello-from
+        image: pbitty/hello-from:latest
```
Comment (sadlil) [inline test/integration/testdata/multinodes/hello-deployment.yaml:28]: Is this a trusted repository? Or would it be better to move this to somewhere else?

**Review by sadlil — CHANGES_REQUESTED** [review]

File: test/integration/multinode_test.go (hunk 2 above)
Comment (hetong07) [reply test/integration/multinode_test.go:446]: According to this [instruction](https://github.com/kubernetes/kubernetes/blob/release-1.2/cluster/addons/dns/README.md#how-do-i-configure-it) use `kubernetes.default`

Comment (hetong07) [reply test/integration/multinode_test.go:446]: According to this [blog](https://medium.com/kubernetes-tutorials/kubernetes-dns-for-services-and-pods-664804211501) decide to use `kubernetes.default.svc.cluster.local`

File: test/integration/testdata/multinodes/hello-deployment.yaml (hunk 4 above)
Comment (hetong07) [reply test/integration/testdata/multinodes/hello-deployment.yaml:28]: This file is directly copied from https://minikube.sigs.k8s.io/docs/tutorials/multi_node/, is it safe? @medyagh

File: test/integration/multinode_test.go (hunk 1 above)
Comment (hetong07) [reply test/integration/multinode_test.go:435]: Why we need this check?

File: test/integration/multinode_test.go [hunk 5]
```diff
@@ -65,6 +66,26 @@ func TestMultiNode(t *testing.T) {
 			})
 		}
 	})
+
+	// sub-test for deploying app.
+	t.Run("serial", func(t *testing.T) {
```
Comment (medyagh) [inline test/integration/multinode_test.go:71]: it is not needed to start another cluster here, DeployApp2Nodes already runs after FreshStart2Nodes and will deplouy on that .

so we u can delete this one

**Review by medyagh — CHANGES_REQUESTED** [review]

**Comment by hetong07** [comment]
@medyagh It seems the multi-node test fails on macOS using virtualbox, when applying config on multinode cluster. The failing reason is that neither pods can resolve to both internal and external address.
```
2021-03-05T00:50:04.4846300Z         -- /stdout --
2021-03-05T00:50:04.4850730Z         ** stderr ** 
2021-03-05T00:50:04.4853040Z         	kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
2021-03-05T00:50:04.4864760Z         	nslookup: can't resolve '(null)': Name does not resolve
2021-03-05T00:50:04.4866450Z         	nslookup: can't resolve 'kubernetes.io': Try again
2021-03-05T00:50:04.4875610Z         	command terminated with exit code 1

```

```
2021-03-05T00:50:15.9023230Z         	kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
2021-03-05T00:50:15.9024800Z         	nslookup: can't resolve '(null)': Name does not resolve
2021-03-05T00:50:15.9026310Z         	nslookup: can't resolve 'kubernetes.default.svc.cluster.local': Try again
2021-03-05T00:50:15.9027290Z         	command terminated with exit code 1
```

Is it flaky? or other reasons?

**Review by medyagh — APPROVED** [review]

**Comment by medyagh** [comment]
thank you @hetong07

**Pull request merged**